pytest_mh/utils/tc.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+from .. import MultihostHost, MultihostUtility
+from ..ssh import SSHLog, SSHProcessError
+
+__all__ = ["LinuxTrafficControl"]
+
+
+class LinuxTrafficControl(MultihostUtility):
+    """
+    Perform traffic control operations on remote host.
+
+    All changes are automatically reverted when a test is finished.
+    """
+
+    def __init__(self, host: MultihostHost) -> None:
+        """
+        :param host: Remote host instance.
+        :type host: MultihostHost
+        """
+        super().__init__(host)
+        self.interface: str = ""
+        self.__restore: set[str] = set()
+        self.__host: MultihostHost = host
+
+    def setup_when_used(self):
+        """
+        Setup traffic control configuration.
+
+        :meta private:
+        """
+        super().setup_when_used()
+
+        # Let's find out the interface that connects the systems
+        result = self.host.ssh.run(f"dig +short {self.__host.hostname}", log_level=SSHLog.Error)
+        try:
+            result = self.host.ssh.run(f"ip -o address | grep {result.stdout}", log_level=SSHLog.Error)
+            self.interface = result.stdout.split(" ")[1]
+        except SSHProcessError:
+            pass
+
+    def teardown_when_used(self):
+        """
+        Revert all traffic control changes.
+
+        :meta private:
+        """
+        tear: str = ""
+        for iter in self.__restore:
+            tear += iter + ";"
+        self.host.ssh.run(tear, log_level=SSHLog.Error)
+        super().teardown_when_used()
+
+    def add_delay(self, time: int, interface: str = ""):
+        """
+        Add delay to network connection.
+
+        :param time: Delay in milliseconds.
+        :type time: int
+        :param interface: Network interface.
+        :type time: ""
+        """
+        if not interface and self.interface is not None:
+            interface = self.interface
+        elif not interface and self.interface is None:
+            raise ValueError(f"Unable to detect the interface for {self.host.hostname}")
```

Comment:
This can be simplified:

```python
if not interface and self.interface is not None:
    interface = self.interface

if not interface:
   raise ...
```